### DISCUSSION TITLE
Issue - POST - net::ERR_EMPTY_RESPONSE

### DISCUSSION THREAD

**Discussion by pmcg86** [cypress-io/cypress#22340] [Questions and Help]
Hi,

Hopefully someone can help me figure out why I have been getting this issue when Cypress tries to login to my website?

First and foremost, this is only an issue through the test runner. I can login to my website via localhost without any issue using a number of different browsers. On a team of 5 people running the same project, it is only happening on my machine.

Project setup:

- All tests are pointing to localhost
- No proxies or anything so just a bog standard Angular front end website with backend project and HeidiSQL DB

Thing to know:

- Cypress has no issue's with reaching the DB as we run scripts to remove data as part of running a test so the first part of the test passes.
- As mentioned above, when project is running, I can login to my website without any issues outside of Cypress

Things I have already tried:

- Tested to see if using a Incognito instance was the issue.
- Removed frontend project, removed Cypress folders including cache folder and pulled project down again.
- Tested if it was only relating to Chromium browsers. Issue exists in Firefox also.
- Set flags in the cypress.json folder for chromeWebSecurity to false.
- Tried different login methods using the UI and also using cy.session
- Checked devDependencies with colleagues to make sure we has all the same versions
- Tried pausing the Test Runner and manually clicking on the Login button.

If anyone has any recommendations or fixes please let me know. If there is any other details that you need I can provide them also.

Image below shows the Console error, this is when I manually click on the Login button

![image](https://user-images.githubusercontent.com/30663900/173884284-fcc23da9-70d6-45ab-bda9-3297f06376c9.png)

**Comment by hlovdal** [comment]
I think I have the same issue, and it is related to how the browser instance is launched.

The Chrome instance launched by Cypress fails to load a localhost url<sup>1</sup> when I open a new tab and enter the url directly, but if I do the same thing on a Chrome window launched as a normal program it works like expected.

<img width="1468" height="645" alt="chrome_localhost" src="https://github.com/user-attachments/assets/2999c4da-ff6a-494d-a002-046db9009673" />


It also fails in the same way for Firefox:

<img width="904" height="396" alt="firefox_localhost" src="https://github.com/user-attachments/assets/53dfe39f-f2a9-49d3-910b-0a2369586a75" />

So that got me investigating a bit further and then started Cypress from a powershell window (which I normally never do). And in that first case the browser instance did not have problems loading the well known url I tested with. However the tests still failed, and subsequent runs now fails loading the test url. So in best case results are inconsistent. Then I also tested starting Cypress from a plain Git Bash window (which I some times do), and behaviour is the same. My initial assumption was that perhaps the environment Git Bash in terminal window in Visual Studio Code (which is the environment I normally use to launch Cypress) was the issue, but I was unable to narrow down to a specific environment.

But it is definitely related to how the browser is launched by Cypress or how the browser behaves when launched by Cypress.

-----------

<sup>1</sup> Replacing `localhost` with `127.0.0.1`<sup>2</sup> works for loading the url, although that is not a viable workaround.

<sup>2</sup> Suggested [here](https://github.com/cypress-io/cypress/discussions/28028) and [here](https://github.com/cypress-io/cypress/issues/5895).

**Comment by hlovdal** [comment]
Possibly the same issue as [this](https://github.com/cypress-io/cypress/issues/6489).
